src/CoreExtensions/Navigation/Builder.php

```diff
@@ -71,7 +71,7 @@ protected function getChildren(Document $parentDocument): array
         foreach ($children as $child) {
             $permissionResource = $child->getProperty('permission_resource');
 
-            if (empty($permissionResource) || $this->currentUser && $this->service->isAllowed($this->currentUser, $child->getProperty('permission_resource'))) {
+            if (empty($permissionResource) && $this->service->isAllowed($this->currentUser, $child->getProperty('permission_resource'))) {
```

Comment:
I would change the phpdoc of the currentUser. It can be null, because it is only set in the setCurrentUser method (not in the construct) and also there only if one exists.

Reply:
@aryaantony92 I would go with the suggested change

Reply:
Done 👍